lib/evmone/instructions.hpp

```diff
@@ -952,6 +961,13 @@ inline code_iterator retf(StackTop /*stack*/, ExecutionState& state, code_iterat
 template <evmc_status_code StatusCode>
 inline StopToken return_impl(StackTop stack, ExecutionState& state) noexcept
 {
+    // RETURN is forbidden inside CREATE3 context
+    if constexpr (StatusCode == EVMC_SUCCESS)
```

Comment:
Why check for this?

Reply:
`stop_impl` has the same branch. All this is a bit confusing, maybe need to document it better.